src/main/java/command/Command.java

```diff
@@ -0,0 +1,82 @@
+package command;
+
+import exception.DukeException;
+import exception.InvalidCommand;
+import exception.InvalidDateFormat;
+import exception.NullDescription;
+import storage.Storage;
+import task.TaskList;
+import ui.Ui;
+
+/**
+ * Command is the over-arching class of all possible commands supported by Duke.
+ * Command classes will provide the instruction on how to handle user inputs.
+ */
+public abstract class Command {
+
+    /**
+     * Parses the input given by users and return the correct type of command.
+     * Invalid commands will throw the respective DukeExceptions
+     *
+     * @param input string input by users
+     * @return The specific type of command that can be executed
+     * @throws InvalidCommand If command cannot be understood
+     * @throws NullDescription If AddCommand is intended but without any description
+     * @throws InvalidDateFormat If AddCommand is intended but the date format is not followed
+     */
+    public static Command parse(String input) throws InvalidCommand, NullDescription, InvalidDateFormat {
+        String[] output = input.split(" ");

```

Comment:
Should the name of the array be outputs?

Reply:
I originally named it output as it was processed the string into individual words. On second thought, I think it might be ambiguous with the new Command that it returns. Thanks for the suggestion!